src/server/routes/response-handler.ts

```diff
@@ -0,0 +1,298 @@
+
+        const query = await MultipleChoiceResponseForm.findById(formID);
+
+        if (choice && query && userID) {
+            if (query.answered.includes(userID)) {
+                res.status(500).end();
+                return;
+            }
+            const currentValue = query.count?.get(choice);
+            if (currentValue !== undefined) {
+                query.count?.set(choice, currentValue + 1);
+                query.answered.push(userID);
+                await query.save();
+                res.status(200).end();
+                return;
+            }
+        }
+        res.status(500).end();
+    })
+);
+
+router.post(
+    "/answerShortAnswer",
+    asyncHandler<
+        undefined,
+        {},
+        {
+            userID: string;
+            userResponse: string;
+            formID: string;
+        }
+    >(async (req, res) => {
+        const formQuery = await ShortAnswerResponseForm.findById(
+            req.body.formID
+        );
+
+        if (formQuery) {
+            if (formQuery.answered.includes(req.body.userID)) {
+                res.status(500).end();
+                return;
+            }
+            const query = await Response.create({
+                userID: req.body.userID,
+                formID: req.body.formID,
+                userResponse: req.body.userResponse,
+            });
+            formQuery.responseID?.push(query.id);
+            formQuery.answered.push(req.body.userID);
+            await formQuery.save();
+            res.status(200).end();
+        } else {
+            res.status(500).end();
+        }
+    })
+);
+
+router.post(
+    "/getMCResults",
+    asyncHandler<Array<Array<string>>, {}, { formID: string }>(
+        async (req, res) => {
+            const query = await MultipleChoiceResponseForm.findById(
+                req.body.formID
+            );
+            if (query) {
+                const count = query.count;
+                const options = query.options;
+
+                if (count && options) {
+                    const dataArray = [
+                        Array.from(count.values()).map(String),
+                        Array.from(options.values()),
+                    ];
+                    res.send(dataArray).status(200).end();
+                } else {
+                    res.status(500).end();
+                }
+            } else {
+                res.status(500).end();
+            }
+        }
+    )
+);
+
+router.post(
+    "/getSAResults",
+    asyncHandler<Array<Array<string>>, {}, { formID: string }>(
+        async (req, res) => {
+            const query = await ShortAnswerResponseForm.findById(
+                req.body.formID
+            );
+            if (query) {
+                if (query.responseID) {
+                    const responseIDs = query.responseID;
+                    let dataArray: Array<Array<string>> = [];
+
+                    for (let id of responseIDs) {
```

Comment:
Map of the list of response ids instead of using a for loop and appending to an array.